package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -414,7 +414,40 @@ export const MessageInputProvider = <
 }>) => {
   const { closePicker, openPicker, selectedPicker, setSelectedPicker } =
     useAttachmentPickerContext();
-  const { client } = useChatContext<StreamChatGenerics>();
+  const { appSettings, client } = useChatContext<StreamChatGenerics>();
+
+  const getFileUploadConfig = (blockedFileOrMime: boolean) => {
+    const fileConfig = appSettings?.app?.file_upload_config;
+    if (fileConfig === null || fileConfig === undefined) {
+      return [];
+    } else {
+      if (blockedFileOrMime) {
+        return fileConfig.blocked_file_extensions;
+      } else {
+        return fileConfig.blocked_mime_types;
+      }
+    }
+  };
+
+  const getImageUploadConfig = (blockedImageOrMime: boolean) => {
+    const imageConfig = appSettings?.app?.image_upload_config;
+    if (imageConfig === null || imageConfig === undefined) {
+      return [];
+    } else {
+      if (blockedImageOrMime) {
+        return imageConfig.blocked_file_extensions;
+      } else {
+        return imageConfig.blocked_mime_types;
+      }
+    }
+  };
```

Comment:
This is a bit confusing. A boolean flag to switch between file extensions and MIME types doesn't make much sense to me, if anything it should've been an enum or similar since `true` and `false` doesn't tell me which is which. Also, a function shouldn't do more than one thing if it can be avoided. Can we simplify this? We could pair code a bit later if you'd like

Reply:
Sure thing

Reply:
✅